skfda/misc/scoring.py

```diff
@@ -826,6 +835,24 @@ def _mse_func(x: EvalPointsType) -> NDArrayFloat:  # noqa: WPS430
 
     return _multioutput_score_basis(y_true, multioutput, _mse_func)
 
+@singledispatch
+def root_mean_squared_error(
+    y_true: DataType,
+    y_pred: DataType,
+    *,
+    sample_weight: NDArrayFloat | None = None,
+    multioutput: MultiOutputType = "uniform_average",
+) -> float | DataType:
+    """Root Mean Squared Error for functional data."""
+
+    return mean_squared_error(
+        y_true,
+        y_pred,
+        sample_weight=sample_weight,
+        multioutput=multioutput,
+        squared=False,  # Apply the square root directly
```

Comment:
Wouldn't this trigger the warning? I think you would need to extract the code (without the warning) to a private function and call it in both cases.